integration-tests/build.rs

```diff
@@ -0,0 +1,247 @@
+    #[error("CompilerOutputDeSerError({0:})")]
+    CompilerOutputDeSerError(String),
+    /// Failed loading Abi from CompactContractRef
+    #[error("ErrorLoadingContractAbi({0:})")]
+    ErrorLoadingContractAbi(String),
+    /// Failed loading Bin from CompactContractRef
+    #[error("ErrorLoadingContractBin({0:})")]
+    ErrorLoadingContractBin(String),
+    /// Failed loading BinRuntime from CompactContractRef
+    #[error("ErrorLoadingContractBinRuntime({0:})")]
+    ErrorLoadingContractBinRuntime(String),
+    /// Failed to convert PathBuff to String
+    #[error("StringConversionError({0:})")]
+    StringConversionError(String),
+    /// Failed to create CompactContractRef from path + name
+    #[error("FailedToLoadCompactContractRef({0:})")]
+    FailedToLoadCompactContractRef(String),
+    /// Failed Bytecode to Bytes conversion
+    #[error("ByteCodeToBytesError({0:})")]
+    ByteCodeToBytesError(String),
+    /// Error serializing json to file
+    #[error("JsonSerializatonError({0:})")]
+    JsonSerializatonError(String),
+    /// Errors from Abigen
+    #[error("GenerateContractBindingsError({0:})")]
+    GenerateContractBindingsError(String),
+}
+
+/// Path to the test contracts
+const CONTRACTS_PATH: &str = "contracts";
+/// Solidity compilation warnings to ignore (by error code)
+/// 2018: Warning - "Function state mutability can be restricted to pure"
+/// 5667: Warning - "Unused function parameter. Remove or comment out the
+/// variable name to silence this warning."
+/// For smart contracts that are optimized for worst case block generation, we want to allow
+/// contracts that do not interfere with state, without setting state mutability to view. otherwise
+/// compiler optimizations will not allow recursive execution of targeted opcodes
+const WARN: &[u64] = &[2018, 5667];
+/// List of contracts as (ContractName, ContractSolidityFile)
+const CONTRACTS: &[(&str, &str)] = &[
+    ("Greeter", "Greeter.sol"),
+    (
+        "OpenZeppelinERC20TestToken",
+        "OpenZeppelinERC20TestToken.sol",
+    ),
+    // Contract to test worst-case usage of opcodes.
+    ("Benchmarks", "BENCHMARKS.sol"),
+];
+/// Target directory for rust contract bingings
+const BINDINGS_DR: &str = "src";
+
+fn main() -> Result<(), BuildError> {
+    println!("cargo:rerun-if-changed=build.rs");
+
+    let solc: Solc = Solc::default();
+    let _solc_version = solc
+        .version()
+        .map_err(|err| BuildError::FailedToGetSolidity(err.to_string()))?;
+
+    for (name, contract_path) in CONTRACTS {
+        let path_sol = Path::new(CONTRACTS_PATH).join(contract_path);
+        let input = CompilerInput::new(&path_sol)
+            .map_err(|err| BuildError::FailedToComposeCompilerInputs {
+                path: path_sol.to_string_lossy().to_string(),
+                reason: err.to_string(),
+            })?
+            .pop()
+            .ok_or_else(|| BuildError::ArtifactError("No Artifact Found".to_string()))?
```

Comment:
The purpose of having a String argument in the error type is to add additional context for debugging. 
For example, we can add the path of the failure or some other error string propagated from the low-level library. The arguments should add more context for the developers to debug.

We use string arguments like "No Artifact Found" in many places in this PR, which doesn't provide additional context than the error type "ArtifactError" itself. 

We can remove string arguments from these error types.

```suggestion
            .ok_or_else(|| BuildError::ArtifactError)?
```

Reply:
Yes that makes sense.